plugin-server/tests/worker/ingestion/timestamps.test.ts

```diff
@@ -108,4 +122,20 @@ describe('parseEventTimestamp()', () => {
         const timestamp = parseEventTimestamp(event)
         expect(timestamp.toUTC().toISO()).toEqual('2020-01-01T12:00:05.050Z')
     })
+
+    it('reports timestamp parsing error and fallbacks to now', () => {
+        const rightNow = DateTime.utc()
+        const event = {
+            team_id: 123,
+            timestamp: 'invalid',
+            now: rightNow,
+        } as any as PluginEvent
+
+        const callbackMock = jest.fn()
+        const timestamp = parseEventTimestamp(event, callbackMock)
+        expect(callbackMock.mock.calls).toEqual([[123]])
+
+        const difference = rightNow.diff(timestamp, 'millisecond').seconds
```

Comment:
we've used freezing time in other places, see https://github.com/PostHog/posthog/blob/2673ce1a40c0b02ded1a59ad60597a7e718351d5/frontend/src/lib/utils.test.ts#L350